sympy/functions/combinatorial/tests/test_comb_factorials.py

```diff
@@ -291,13 +291,14 @@ def test_binomial():
     assert binomial(10, 10) == 1
     assert binomial(n, z) == 1
     assert binomial(1, 2) == 0
+    assert binomial(-1, 2) == 1
     assert binomial(1, -1) == 0
     assert binomial(-1, 1) == -1
-    assert binomial(-1, -1) == 1
+    assert binomial(-1, -1) == 0
```

Comment:
Not sure about such scenario but this seems a special case (nCn should be 1, no matter what are the values of n). Priority should be given to test such case ((n-k)=0) first and then check if `k` is negative (and if so return 0).
@jksuom would rather give a better insight into this situation.

Reply:
Actually this is in accordance with the [definition 3](https://www.johndcook.com/blog/binomial_coefficients/) mentioned by @normalhuman.
The order of limits should define this result as `0`, I think.

>  Priority should be given to test such case ((n-k)=0)

Yes, I thought about this. But I am not sure if its good to convert `C(n, k)` to `C(n, n-k)` if we are using [definition 3](https://www.johndcook.com/blog/binomial_coefficients/) otherwise it might create a cycle in this [flow chart](https://www.johndcook.com/blog/computing_binomial_coefficients/).

Reply:
It is possible to express `binomial(n, k)` in terms of gamma functions as a meromorphic function of two variables. A meromorphic function of a single variable always has a limit (which may be `zoo`) but that is not the case with functions of several variables. The limit may exist only if the variables are somehow restricted.

In SymPy, the definition of `binomial(n, k)` is restricted by demanding that `k` be an integer. So the natural way of computing the limit should be setting the `k` value first. (If we set `n = k`, then also `n` should be restricted to integer values and it does not make sense to speak of a limit.) As a meromorphic function of a single variable `k` (for non-integer but constant `n`) `binomial(n, k)` (rewritten in terms of gamma functions) has the well defined value 0 for all negative integers `k`. That it also the limit value of `binomial(n, k)` when `n` tends to an integer.

Reply:
Thank you for the explanation.

> As a meromorphic function of a single variable k (for non-integer but constant n) binomial(n, k) (rewritten in terms of gamma functions) has the well defined value 0 for all negative integers k. That it also the limit value of binomial(n, k) when n tends to an integer.

I think this makes it clear.